tutorials/nemotron-cc/nemotron_cc.ipynb

```diff
@@ -0,0 +1,4536 @@
+       "    <tr>\n",
+       "      <th>4</th>\n",
+       "      <td>009b7a3794bce22726148ac37b74b92b</td>\n",
+       "      <td>[EN_CC-0000059195, EN_CC-0000059196, EN_CC-000...</td>\n",
+       "    </tr>\n",
+       "  </tbody>\n",
+       "</table>\n",
+       "</div>"
+      ],
+      "text/plain": [
+       "                            _hashes  \\\n",
+       "0  0016fa4543e1d6913cf7ebebe9b2d2f9   \n",
+       "1  00246b1ca1ac91e27c0e9d5806f6fbf2   \n",
+       "2  00362d462942216e6aa8fcc583c2ca81   \n",
+       "3  0090d246b4f44804647d2601964a2932   \n",
+       "4  009b7a3794bce22726148ac37b74b92b   \n",
+       "\n",
+       "                                                  id  \n",
+       "0               [EN_CC-0000014715, EN_CC-0000044369]  \n",
+       "1               [EN_CC-0000022205, EN_CC-0000036878]  \n",
+       "2               [EN_CC-0000051215, EN_CC-0000066095]  \n",
+       "3  [EN_CC-0000014830, EN_CC-0000059283, EN_CC-000...  \n",
+       "4  [EN_CC-0000059195, EN_CC-0000059196, EN_CC-000...  "
+      ]
+     },
+     "execution_count": 19,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "exact_dedup_res = pd.read_parquet(os.path.join(exact_duplicates_dir,\"_exact_duplicates.parquet\"))\n",
+    "print(f\"Number of exact duplicated document:{len(exact_dedup_res)}\")\n",
+    "duplicated_list = exact_dedup_res.groupby(\"_hashes\")[\"id\"].agg(list).reset_index().head()\n",
+    "duplicated_list"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "5e5d01fe",
+   "metadata": {},
+   "source": [
+    "Remove duplicates from original dataset."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 20,
+   "id": "a7b9fbb9",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Reading 1 files with blocksize='1gb' / files_per_partition=None\n"
+     ]
+    },
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "/opt/NeMo-Curator/nemo_curator/utils/distributed_utils.py:450: UserWarning: If underlying Parquet data does not have consistent schema, reading with blocksize will fail. Please update underlying RAPIDS package to version 25.02 or higher, or use files_per_partition approach instead.\n",
+      "  warnings.warn(\n"
+     ]
+    }
+   ],
+   "source": [
+    "exact_duplicates = DocumentDataset.read_parquet(os.path.join(exact_duplicates_dir,\"_exact_duplicates.parquet\"), backend=\"cudf\")\n",
+    "exact_docs_to_remove = exact_duplicates.df.map_partitions(\n",
+    "    lambda x: x[x._hashes.duplicated(keep=\"first\")]\n",
+    ")\n",
+    "input_id_field = \"id\"\n",
+    "#Remove the duplicated document from input dataset\n",
+    "result = input_dataset.df[\n",
+    "    ~input_dataset.df[input_id_field].isin(exact_docs_to_remove[input_id_field].compute())\n",
+    "]\n",
+    "result.to_parquet(exact_dedup_output_dir, write_to_filename=True)"
```

Comment:
Can replace this custom code with the provided `remove` methods in the deduplication classes.